src/gui/statusbar.cpp

```diff
@@ -66,7 +66,7 @@ StatusBar::StatusBar(QWidget *parent)
     m_connecStatusLblIcon->setIcon(UIThemeManager::instance()->getIcon(QLatin1String("firewalled")));
     m_connecStatusLblIcon->setToolTip(
         QString::fromLatin1("<b>%1</b><br><i>%2</i>").arg(tr("Connection status:")
-            , tr("No direct connections. This may indicate network configuration problems.")));
+            , tr("No incoming connections.\nIf you have a dedicated IP address, this may indicate a problem with your network configuration.")));
```

Comment:
I find a bit confusing, could you elaborate more? What does it mean to have a "dedicated IP address"? Is qbt setup to listen on only one interface? or does it mean the machine has only 1 ip address?

Reply:
I think this can confuse some users. Why not simply use this? =

> No incoming connections.\nThis may indicate a problem with your network configuration.

Reply:
There was a bug in older libtorrent(used in qbittorrent versions 4.3.3 and lower) where incoming packets received through DHT could be mistaken for incoming connection and therefore green plug icon was displayed for almost all users(unless they turned off DHT). Now after that bug got patched people started seeing the orange plug icon(in 4.3.4 onwards). 
Most of these are shared IP users. The aim of this PR is not make it more obvious that orange plug doesn't mean anything wrong unless they have a real IP.

Reply:
@an0n666 

I think something like the following is clearer: `No incoming connections.\nYou either have a shared IP address, or you may have a problem with your network configuration.`

It is more google-able because the user may not know what is the opposite of a "dedicated IP address".

Also if it doesn't make the tooltip much bigger, I would add `(...)may have a problem with your network configuration. Most often it's one of the following: firewall or port forwarding issues/misconfiguration, bad UPnP/NAT-PMP support by the router.`

Reply:
>AFAIK the orange status still indicates a connection problem: outside peers cannot reach qbt and therefore qbt cannot upload to them. I.e. qbt transfer capabilities is crippled, isn't that the case?
>
Yes, that’s correct and expected on a shared IP. Because there’s no way to fix this problem on shared IP connections. Only the ISP can forward a port for the user if they wanted to.

>incorrect upnp setup for a shared IP user.
>
Shared IP users are usually behind double NATs. One is ISP router NAT and another one is home router NAT. 
UPnP/port forwarding doesn’t work in such situations because you’d have to forward the ports in both NATs. So incorrect UPnP/firewall settings doesn’t play any roles there.

Reply:
@Chocobo1

Good points. Perhaps I am too optimistic in thinking that a casual user will want to google anything at all about network topology :). I agree with your proposed change to the tooltip.

With that said, I still think we should still try to not send users who have no hope of fixing the problem on their own (users on shared IPs behind double NATs, like @an0n666 mentions) on a wild goose chase. I imagine that after the user finds that nothing is wrong with the usual suspects (`firewall or port forwarding issues/misconfiguration, bad UPnP/NAT-PMP support by the router.`), they would likely search for help on the wiki and even ultimately open an issue about that here.

To mitigate this issue, I propose at least adding an entry to the FAQs on the wiki about this. @an0n666 do you know of any simple online service that the users can use to find out if their are in such a situation? Then the FAQ entry could say something like:

```
### Everything is fine with my firewall/port forwarding and UPnP/NAT-PMP router support, but I still get the "No direct connections"/orange flame icon warning.

You could be on a shared IP behind a double NAT. In that case, nothing can be done. You can check if this is the case here: https://FOO.BAR.
```

We could then also add a link to the FAQs to the tooltip, but the tooltip is probably already too big for that as it is.

Reply:
Provide an IP detection website and tell users that if their router's WAN IP is different from their public IP then they are behind CG-NAT? Or Wikipedia's entry about private IP address?

Reply:
>Shared IP users are usually behind double NATs. One is ISP router NAT and another one is home router NAT.
UPnP/port forwarding doesn’t work in such situations because you’d have to forward the ports in both NATs. So incorrect UPnP/firewall settings doesn’t play any roles there.

Thanks for the explanation I get a better picture now.
Currently I think helping the users on this matter isn't an easy task with just a line or two explanation in the tooltips. As it would possibly misguide users of different network setups.
IMO it would be better, like others proposed, to have a link to a wiki entry (faq and the like) and provide detailed steps to diagnosis to real cause.

>Provide an IP detection website and tell users that if their router's WAN IP is different from their public IP then they are behind CG-NAT?

I didn't find anything on google...
Just to be clear, such detection would be very useful but it should only be listed on wiki entry and not mentioned in qbt program to avoid user-tracking concerns/issues and affiliation concerns with the service provider.

Reply:
As a casual user, just how does this double-NAT thing apply to those of us using a vpn?